### PR TITLE
Move Quote related Plugins to correct module

### PR DESCRIPTION
**Pull request by sankalpshekhar**
### Description (*)

Move plugins to correct module in Magento_Quote
### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-5234](https://jira.corp.magento.com/browse/ENGCOM-5234) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Comment by ysapiga** [comment]
@sankalpshekhar please fix static tests fail.
https://testing-service.magento-community.engineering/reports/magento/magento2/pull/23174/99984a56-573c-41a5-a7aa-1df03c10f1f2/14636/Statics/allure-report-ce/index.html#suites/ab1a8c7bdd131285558c80e8ade1655f/33805a79466c4eb1/

**Comment by sankalpshekhar** [comment]
@ysapiga I don't the static test results are correct, as it referring to get a different name for the plugin.

Had this been an issue functionally, the QA tests should have failed, which isn't the case.

File: app/code/Magento/Quote/etc/webapi_rest/di.xml [hunk 1]
```diff
@@ -12,4 +12,7 @@
     <type name="Magento\Quote\Model\QuoteRepository">
         <plugin name="accessControl" type="Magento\Quote\Model\QuoteRepository\Plugin\AccessChangeQuoteControl" />
     </type>
+    <type name="Magento\Quote\Model\QuoteRepository">
+        <plugin name="authorization" type="Magento\Quote\Model\QuoteRepository\Plugin\Authorization" />
```
Comment (sivaschenko) [inline app/code/Magento/Quote/etc/webapi_rest/di.xml:14]: Please add this plugin declaration to an existing `Magento\Quote\Model\QuoteRepository` type node in this file

File: app/code/Magento/Quote/etc/webapi_soap/di.xml [hunk 2]
```diff
@@ -12,4 +12,7 @@
     <type name="Magento\Quote\Model\QuoteRepository">
         <plugin name="accessControl" type="Magento\Quote\Model\QuoteRepository\Plugin\AccessChangeQuoteControl" />
     </type>
+    <type name="Magento\Quote\Model\QuoteRepository">
+        <plugin name="authorization" type="Magento\Quote\Model\QuoteRepository\Plugin\Authorization" />
```
Comment (sivaschenko) [inline app/code/Magento/Quote/etc/webapi_soap/di.xml:14]: Please add this plugin declaration to an existing `Magento\Quote\Model\QuoteRepository` type node in this file

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @sankalpshekhar thanks for the contribution! Please take a look at my review comments

**Review by sivaschenko — APPROVED** [review]
Code review comment and static tests are already fixed internally.

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-5234](https://jira.corp.magento.com/browse/ENGCOM-5234) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by sankalpshekhar** [comment]
@sivaschenko  I have made the requested changes, and PR has been updated.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-5299](https://jira.corp.magento.com/browse/ENGCOM-5299) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**